xc/xenops_server_xen.ml

```diff
@@ -1460,6 +1460,7 @@ module VM = struct
               match vbd.Vbd.ty, vbd.mode with
               | Vbd.Disk, ReadOnly -> None
               | Vbd.Disk, _        -> Some (index, path, Device.Dm.Disk)
+              | Vbd.Floppy, _      -> Some (index, path, Device.Dm.Floppy)
               | _                  -> Some (index, path, Device.Dm.Cdrom)
```

Comment:
I'd replace the wildcard with Vbd.Cdrom to be explicit and avoid treating a floppy as a cdrom by accident.